### ISSUE TITLE
cannot customize some faces of org-transclusion 

### ISSUE THREAD

**Issue by ffsammak** [nobiot/org-transclusion#174]
I can see and get the following faces customized by M-x customize-faces and list-faces-display:

![Screenshot from 2023-03-06 21-13-06](https://user-images.githubusercontent.com/64158105/223216598-e0a89c08-d00e-49a5-b114-0c37ace88676.png)

org-transclusion-live-sync-start works fine and shows the corresponding custom-faces

![Screenshot from 2023-03-06 21-23-51](https://user-images.githubusercontent.com/64158105/223216960-fc8eb424-e8b5-4c4a-a05e-040ae040627c.png)

However, for the faces below I cannot see anything changing no matter how I do customizations to them:
 - org-transclusion
 - org-transclusion-source
 - org-transclusion-keyword 

![Screenshot from 2023-03-06 21-13-36](https://user-images.githubusercontent.com/64158105/223217354-5f3b4b28-62e3-494f-ab83-810e5ffe325b.png)

I am using vanilla Emacs and started to debug this issue by running emacs -Q but to no avail. I use straight for package installation and use-package for their configuration. 

So in summary, all custom-faces are fine except: org-transclusion, org-transclusion-source, and org-transclusion-keyword they don't get changed. Is this a known limitation, or am I missing something? I tried to progressively comment packages/code to identify any interfering setup code in my init.el but to no avail. Any help would be appreciated.

- Emacs: GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.33, cairo version 1.16.0)
 of 2023-03-02
- Framework: vanilla emacs
- Org: Org mode version 9.6.1
- Org-transclusion: Version: 1.3.2 on 05 March 2023

**Comment by nobiot** [comment]
Oh, right. Thanks for reporting. It looks like I have forgotten to put `org-transclusion` and `org-transclusion-source` back (there was a design change, if I remember correctly). Let me take some time do so. 

`org-transclusion-keyword` works, though. See the color changed for `#+transclude:` keyword in the buffer to a green. You might need to call `revert-buffer` in order for the keyword to take the Org-transclusion default.
![image](https://user-images.githubusercontent.com/12507865/223224211-8def7547-868d-4147-a9b8-b5bea00b2d54.png)

**Comment by nobiot** [comment]
The latest commit has add the faces back in. If you are still around, it would be great if you could spend some time to do a quick test. The commit should make its way to [GNU-devel repository](https://elpa.gnu.org/devel/org-transclusion.html) overnight. 

Absolutely fine if you don't have time; wanted to let you know anyway. Thanks for reporting the issue; it helps improve the package for real usage.

**Comment by nobiot** [comment]
Thank you for taking the time for reporting back; much appreciated.

> I am trying the one that starts with [0273975](https://github.com/nobiot/org-transclusion/commit/02739759311c70a2eb09d19b4261c4441018acd7). If you meant a different one, please let me know.

Yes, it is a good commit to test. Apologies, the commit I was referring to was linked and appear in this thread in GitHub website, so I thought it would be enough. Perhaps you prefer to work with email for this type of communication and didn't see the linked commit. I will try to be clear next time. Thank you.

The commit was actually only about the topic of this thread: "cannot customize some faces of org-transclusion". Thank you for noting other observations. I will respond to each one below.


#### Faces

> * org-transclusion: The face didn't change. Is this the transcluded text? If yes, then it didn't bother to change.
> * org-transclusion-source: The face didn't change.

These are the two faces that you originally reported, which I added back in. In order not to influence other users, the default is "transparent" -- only the face definition with empty specification. You'd have to add color and other specifications on your end to see the effect.

I will make this clear in the user manual and document string for these two faces.


#### `org-transclusion-fringe`

> * org-transclusion-fringe (I had to move the #+transclude keyword a bit to get it to show up, it sometimes disappears when I do revert-buffer, hence buggy)

First time I hear about this issue. If you could reproduce it reliably, I'd appreciate your opening a new issue -- I have never had any issue with the fringe disappearing. I will take note of it and try doing `revert-buffer` and see if I can reproduce it.

#### Live-sync

> * another issue I observed is that live-sync (bound to e) doesn't start when I am on the org header text, but it works fine in the normal text under it. I think I didn't have this issue in the older version. `#+transclude: [node][Description]] :level 2`

I will try to reproduce it. If I can identify the issue, I will open a new issue as a log. Let me see. One of the recent commits might have introduced a regression issue.


#### infinite loop

> Note: when I fiddle around with the transclude region and hit save the buffer, it will take forever and not get saved. I realized then that the buffer is creating multiples of the #+transclude line, and the buffer size is getting bigger and bigger with time. I noticed this issue in the previous versions as well. By the time I canceled saving, I had too many multiples of transclude lines to deal with and delete. My last resort, a poor man's solution, was to delete the whole buffer or close emacs altogether, which was unfortunate sometimes.

This has been reported #177. It's something I have been trying to reproduce for a long time -- I have had this issue **only when** I am developing the code, so it's expected then -- after the release of the code, I have never been able to have this problem. 

As I cannot identify the root cause, I am currently trying to put some heuristics to detect this issue and error to break the loop. If you can reliably reproduce the problem, I'd appreciate it to add the procedure in issue #177.

**Comment by nobiot** [comment]
Added some documentation for "transparent" default face specification.

eecb77b

**Comment by nobiot** [comment]
“Transparent” simply means there is no specification (no color) by default. At a quick glance, the screen looks good to me. Does it not do anything?

**Comment by nobiot** [comment]
That one is for the source. So won’t change the transclusion, though. Transclusion has its own face.

**Comment by nobiot** [comment]
Okay, thank. I am looking. I may be seeing something wrong with the latest codebase... Maybe I made a mess with git... Let me check.

**Comment by nobiot** [comment]
Okay, definitely something went wrong with codebase change. Thank you for quickly returning to my request. I will look more carefully later today (probably evening in my timezone).

**Comment by nobiot** [comment]
The new commit [bc24be2](https://github.com/nobiot/org-transclusion/commit/bc24be23348a6849868b6c49b865e56a175b50c5) should fix the issue....

The two faces 'org-transclusion' and 'org-transclusion-source' would still need to be customized to add any shade of colors, like these below:

![image](https://github.com/nobiot/org-transclusion/assets/12507865/83791f67-af40-4af7-bc80-781502c6cb43)

![image](https://github.com/nobiot/org-transclusion/assets/12507865/767c6929-7813-4d5d-901e-c88fb6f679b2)

**Comment by nobiot** [comment]
@ffsammak Thank you for confirming :)
